### PR TITLE
Unitful ticks and lims

### PR DESCRIPTION
**Pull request by gustaphe**
Solves an issue from UnitfulRecipes: https://github.com/jw3126/UnitfulRecipes.jl/issues/82#issuecomment-1264380067

```julia
plot(randn(5)u"m")
plot!(; ylims=(0u"cm", 5u"cm"), yticks=[1u"cm", 2u"cm", 3u"cm"])
```

### CODE REVIEW

**Review by t-bltg — APPROVED** [review]
LGTM, pending CI.
You have merge rights now - if I'm not mistaken - so you can merge at will.

**Comment by t-bltg** [comment]
We have a formatter in `Plots`, you need to run:
```julia
using JuliaFormatter
format(["src", "test"])
```
for the format check to pass.

File: Project.toml
Comment (t-bltg) [inline Project.toml:4]: We usually prefer to push a dedicated commit with the ver num in the commit text for version bump (this help visually inspecting commits).

Comment (gustaphe) [reply Project.toml:4]: oh, okay. I'll do some force push magic once CI is happy.

Comment (t-bltg) [reply Project.toml:4]: If you do it now, you'll spare waiting for CI twice ;)

Comment (t-bltg) [reply Project.toml:4]: The policy here in `Plots` is to squash `PRs` before merging, again for clarity.
So what I meant was remove the version bump in this `PR`, merge when `CI` is green, and then edit `Project.toml` in the github interface to only increase the version, with the version in the title commit as e.g. https://github.com/JuliaPlots/Plots.jl/commit/87d2470df3203e378f43c6c2a877789c2d0e2a24.
I wasn't clear enough on that one (you'll know for the next `PR`, no need to change it now).

**Pull request merged**
